ext/standard/basic_functions.c

```diff
@@ -2505,7 +2505,7 @@ static void php_simple_ini_parser_cb(zval *arg1, zval *arg2, zval *arg3, int cal
 			}
 
 			if (!(Z_STRLEN_P(arg1) > 1 && Z_STRVAL_P(arg1)[0] == '0') && is_numeric_string(Z_STRVAL_P(arg1), Z_STRLEN_P(arg1), NULL, NULL, 0) == IS_LONG) {
-				zend_ulong key = (zend_ulong) zend_atol(Z_STRVAL_P(arg1), Z_STRLEN_P(arg1));
+				zend_ulong key = (zend_ulong) ZEND_STRTOUL(Z_STRVAL_P(arg1), NULL, 0);
```

Comment:
This parses ini entries like this:

```
1[]=2
```

The behavior is not changed here, because we enter this branch only when arg1 is a numeric string

Reply:
Maybe add this as a comment?

Reply:
Do you mean the fact that we enter this branch when arg1 is numeric string ?

Reply:
Yes :)